### PR TITLE
Handle files with a variety of encodings & newline styles

### PR DESCRIPTION
**Pull request by aelkiss**
After running into issues with IO.popen we decided to break it down into basic parts. A compressed file is simply `zcat`ed behind the scenes, and BOM and newline stuff is handled in `File.open`.

### CODE REVIEW

**Review by mwarin — APPROVED** [review]
Looks good. 
Did you write any tests for `Utils::OpenWhatever`? If so you did not push them.

**Comment by aelkiss** [comment]
I did not. I think it's worth extracting the tests for encoding & newline style and moving them to tests for this little class.

File: lib/utils/open_whatever.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#
+#  It can handle:
+#   * UTF-8 (including plain ASCII) with or without BOM
+#   * UTF-16LE or UTF-16BE (if BOM is present)
+#   * CR, LF, or CRLF newlines.
+#   * gzipped files, if they end in .gz
+#   * uncompressed files
+# Usage:
+#
+# Utils::LineCounter.new(some_file).count_lines # -> number of lines (as an int)
```
Comment (aelkiss) [inline lib/utils/open_whatever.rb:14]: pure copypasta right there...

**Comment by aelkiss** [comment]
This will supercede #261 and #262 when it's done.

**Comment by aelkiss** [comment]
Conclusion: This doesn't work, everything is awful, `IO.popen` won't handle the byte order markers; also it all can't be made to work with UTF-16.

**Review by aelkiss — COMMENTED** [review]
Looks fine except we should remove `OpenWhatever`.

**Comment by aelkiss** [comment]
I can't approve my "own" PR, but this looks good to me now :+1:

**Comment by mwarin** [comment]
And I already approved it back when it was your PR. What a mess. 
Here we go!
>merge<

**Pull request merged**